### ISSUE TITLE
How to fix "MissingEntitlement: SecKeychainItemCopyContent refused to return data"?

### ISSUE THREAD

**Issue by hktonylee** [iqlusioninc/keychain-services-rs#22]
I tried to run the following program but it panic with: `"Error { kind: MissingEntitlement, backtrace: , description: "SecKeychainItemCopyContent refused to return data" }"`

```rust
extern crate keychain_services;

use keychain_services::{Keychain, keychain};

fn main() {
    println!("Getting keychain...");
    if let Ok(default_keychain) = Keychain::find_default() {
        println!("Getting password...");
        if let Ok(generic_password)
            = keychain::item::GenericPassword::find(&default_keychain, "TestTest", "user") {
            println!("Printing password...");
            println!("Password: {}", generic_password.password().unwrap().as_str())
        }
    }
}
```

I tried `codesign` as instructed in the [Apple Developer guide](https://developer.apple.com/library/archive/documentation/Security/Conceptual/CodeSigningGuide/Procedures/Procedures.html). I created a self-signed certificate. Then I used [gimmedebugah](https://github.com/gdbinit/gimmedebugah) to inject `Info.plist` to the binary. Lastly I run the `codesign` using the self-signed certificate. I can verify the app is correctly signed with the following command

```
> codesign --verify --deep --strict --verbose=2 my_app
my_app: valid on disk
my_app: satisfies its Designated Requirement
```

In the `Info.plist` injection step, I also tried to linker flag `-sectcreate __TEXT __info_plist Info.plist -o my_app.out` to the `.cargo/config`.

But the app still cannot retrieve password from keychain. Now I have no clue what to do next. Would you mind sharing any step-by-step guideline? It will be very useful!

**Comment by hktonylee** [comment]
It is probably the place where the error is thrown: https://github.com/iqlusioninc/keychain-services.rs/blob/f5d6e4bad8df8ff400abc05cc497ecea0ceda474/src/keychain/item/mod.rs#L59-L62

Actually I succeeded to get password without any codesign using another library: https://github.com/kornelski/rust-security-framework. Therefore I think this error is not caused by codesign. I suggest other people not to dig into this problem anymore.

**Comment by tarcieri** [comment]
@hktonylee now that `rust-security-framework` is maintained again, it's probably the way to go, especially as it's already working for you

**Comment by hktonylee** [comment]
@tarcieri Yes that's true. Although that already spent me several hours :-/
